src/components/shared/formItems/ProjectHandle.tsx

```diff
@@ -17,29 +17,32 @@ export default function ProjectHandle({
   hideLabel,
   formItemProps,
   onValueChange,
-  value,
+  initialValue,
   requireState,
   returnValue,
 }: {
-  onValueChange: (val: string) => void
-  value?: string | BigNumber
-  requireState?: 'exists' | 'notExist'
+  onValueChange?: (val: string) => void
+  initialValue?: string | BigNumber
+  requireState?: 'exists' | 'notExist' // whether the handle is required to already exist or not.
   returnValue?: 'id' | 'handle'
 } & FormItemExt) {
   const {
     theme: { colors },
   } = useContext(ThemeContext)
   const [inputContents, setInputContents] = useState<string>()
-
+  const [handleLoading, setHandleLoading] = useState<boolean>(false)
   const { contracts } = useContext(UserContext)
 
-  // Value can be either a project ID, or a project's handle
+  // initialValue can be either a project ID, or a project's handle
   useEffect(() => {
-    if (typeof value === 'string') {
-      setInputContents(value)
-    } else if (isBigNumberish(value)) {
+    if (initialValue) {
+      setHandleLoading(true)
```

Comment:
If initialValue is truthy, display the spinner. `idForHandle`'s callback will remove the spinner in its callback fn.